packages/renderer/src/lib/preferences/PreferencesConnectionDetailsTerminal.svelte

```diff
@@ -25,15 +25,15 @@ interface ProviderDetailsTerminalProps {
 let { provider, connectionInfo, screenReaderMode = false }: ProviderDetailsTerminalProps = $props();
 let terminalXtermDiv: HTMLDivElement;
 let shellTerminal: Terminal;
-let currentRouterPath: string;
 let sendCallbackId: number | undefined;
 let terminalContent: string = '';
 let serializeAddon: SerializeAddon;
 let lastState = $state<ProviderConnectionStatus>('started');
+let terminalOnDataListener: IDisposable | undefined = undefined;
 
 $effect(() => {
   const connectionStatus = connectionInfo.status;
-  if (lastState === 'stopped' && connectionStatus === 'started') {
+  if (lastState === 'starting' && connectionStatus === 'started') {
```

Comment:
I would keep the `stopped` one, which can be useful in other cases (I can see this component is not limited to VMs).
Perhaps `connectionStatus === 'started' && connectionStatus !== lastState` ?